source/03-components/table/table--sortable.es6.ts

```diff
@@ -0,0 +1,295 @@
+  header: HTMLTableCellElement,
+  isAscending: boolean
+): void => {
+  const table = header.closest(TABLE) as HTMLTableElement;
+  let safeAscending = isAscending;
+
+  if (typeof safeAscending !== 'boolean') {
+    safeAscending = header.getAttribute(SORTED) === ASCENDING;
+  }
+
+  if (!table) {
+    throw new Error(`${SORTABLE_HEADER} is missing outer ${TABLE}`);
+  }
+
+  safeAscending = sortRows(header, isAscending);
+
+  if (safeAscending) {
+    getColumnHeaders(table).forEach((otherHeader: HTMLTableCellElement) => {
+      if (otherHeader !== header) {
+        unsetSort(otherHeader);
+      }
+    });
+    updateLiveRegion(table, header);
+  }
+};
+
+/**
+ ** Inserts a button with icon inside a sortable header.
+ * @param {HTMLTableCellElement} header
+ * @param {string} imagePath
+ */
+
+const createHeaderButton = (
+  header: HTMLTableCellElement,
+  imagePath: string
+): void => {
+  const buttonEl = document.createElement('button');
+  buttonEl.setAttribute('tabindex', '0');
+  buttonEl.classList.add(SORT_BUTTON_CLASS);
+
+  // Add button icons.
+  buttonEl.innerHTML = `
+        <svg class="c-icon c-table__sort-icon"><use xlink:href="${imagePath}/sprite.artifact.svg#sort"></use></svg>
+        <svg class="c-icon c-table__sorted-icon"><use xlink:href="${imagePath}/sprite.artifact.svg#sorted"></use></svg>
+      `;
+  const headerInnerElem = document.createElement('div');
+  headerInnerElem.innerHTML = header.innerHTML;
+  headerInnerElem.classList.add('c-table__header-inner');
+  headerInnerElem.appendChild(buttonEl);
+  header.innerHTML = '';
+  header.appendChild(headerInnerElem);
+  updateSortLabel(header);
+};
+
+Drupal.behaviors.sortableTable = {
+  attach(context, settings) {
+    const { imagePath } = settings.gesso;
+
+    const tables = once(
+      'sort-buttons',
+      '.c-table.is-sortable',
+      context
+    ) as NodeListOf<HTMLElement>;
+
+    tables.forEach(table => {
+      // Add sortable attribute to all column headers if wanted.
+      if (SORT_ALL_COLUMNS) {
+        table.querySelectorAll<HTMLTableCellElement>('thead th').forEach(th => {
+          th.setAttribute(SORTABLE_ATTRIBUTE, '');
+        });
+      }
+
+      // Add aria-live region for sort announcements.
+      const liveRegion = table.nextElementSibling as HTMLElement;
+      if (!liveRegion || !liveRegion.matches(ANNOUNCEMENT_REGION)) {
+        table.insertAdjacentHTML(
+          'afterend',
+          `<div class="${ANNOUNCEMENT_CLASS}" aria-live="polite"></div>`
+        );
+      }
+
+      const sortableHeaders =
+        table.querySelectorAll<HTMLTableCellElement>(SORTABLE_HEADER);
+      sortableHeaders.forEach(header => {
+        createHeaderButton(header, imagePath || '');
+      });
+
+      const sortButtons: NodeListOf<HTMLButtonElement> =
+        table.querySelectorAll(SORT_BUTTON);
+      sortButtons.forEach(button => {
+        button.addEventListener('click', event => {
+          const target = event.target as HTMLElement;
```

Comment:
Remove casting and do a check:
```ts
const target = event.target;
if (!(target instanceof HTMLElement)) {
  return;
}
```

Reply:
@mdrayer 
You suggested change resulted in a `Use object destructuring` linting error.

Updating to the following broke the script, but didn’t result in any errors.

```js
const { target } = event;

  if (!(target instanceof HTMLElement)) {
    return;
  }
```

Reply:
Hmm, what broke with the script? Maybe `HTMLElement` isn't the right type here. Could you try `Element` instead?
```ts
const { target } = event;
if (!(target instanceof Element)) {
  return;
}
```

Reply:
Or `HTMLButtonElement` since that's how `sortButtons` are typed?

Reply:
@kmonahan I am getting a spread of different elements on `event.target` on this event:
- `<svg class="c-icon c-table__sort-icon">`
- `<button tabindex="0" class="c-table__header-button">`
- `<use xlink:href="images/sprite.artifact.svg#sorted></use>`

I'd guess that sticking with the `Element` check is the way to go here, unless we want to either adjust the event target or use a different property of `event` for the check here.

Reply:
Whatever @dcmouyard thinks makes sense here -- I think `Element` is fine too

Reply:
Ah yep, `currentTarget` works as well here and is consistently giving me the `<button>` element.

Reply:
Thanks for your help! Got it working with
```js
const target = event.currentTarget;
if (!(target instanceof HTMLButtonElement)) {
  return;
}
```